### PR TITLE
intersection between sphere, line and plane, circle

### PR DESCRIPTION
**Pull request by romanarust**
Added `intersection_sphere_line` and `intersection_plane_circle`

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [X] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.

### Checklist

1. [X] Add the change to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading: `Added`, `Changed`, `Removed`.
1. [ ] Run all tests on your computer (i.e. `invoke test`).
1. [ ] If you add new functions/classes, check that:
   1. [ ] Are available on a second-level import, e.g. `compas.datastructures.Mesh`.
   1. [ ] Add unit tests (especially important for algorithm implementations).

### CODE REVIEW

**Comment by gonzalocasas** [comment]
I can't really judge the math, but the code looks good to me.
I would however add a couple of unit tests if possible.

**Review by gonzalocasas — APPROVED** [review]
image

**Comment by romanarust** [comment]
friendly ping to @brgcode. 
Could you please have a look if it is ok to create `tangents.py` in `compas.geometry._core`. I didn't know where else to put the function `tangent_points_to_circle_xy`. I thought similar functions like `tangent_plane_to_sphere` could end up there.

File: src/compas/geometry/_core/tangent.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import math
+
+
+def tangent_points_to_circle_xy(circle, point):
+    """Calculates the tangent points on a circle in the XY plane.
```
Comment (tomvanmele) [inline src/compas/geometry/_core/tangent.py:5]: do i understand correctly that you compute the points on the circle that create tangent lines from the given point?

Comment (romanarust) [reply src/compas/geometry/_core/tangent.py:5]: yes

File: src/compas/geometry/_core/intersections.py [hunk 2]
```diff
@@ -734,11 +736,117 @@ def intersection_segment_polyline_xy(segment, polyline, tol=1e-6):
...
+        center, radius of the sphere.
+    line : tuple
+        xyz coordinates of two points defining the line.
+
+    Returns
+    -------
+    None
+        If there is no intersection.
+    point : tuple
+        One intersection point.
```
Comment (tomvanmele) [inline src/compas/geometry/_core/intersections.py:759]: one intersection means the line is tangent to the sphere?

Comment (tomvanmele) [reply src/compas/geometry/_core/intersections.py:759]: ha sorry, just saw it in the extended summary. perhaps add it here as well?

**Comment by tomvanmele** [comment]
@romanarust `tangents.py` seems entirely appropriate...

**Comment by romanarust** [comment]
@gonzalocasas, @brgcode thank you!

**Pull request merged**